solve #7 and clean up failed test

wgx731 opened the pull request:
Hi @janstenpickle,

Really nice work on `scala-vault`, but `vault` has changed a lot until now(e.g issue #7). Therefore I have done some maintenance work to make `scala-vault` work with `vault` version `0.6.5`. Please help review the changes and publish a newer version if possible.

By the way, circle CI doesn't really have good docker support (e.g. docker is still at version 1.9), therefore I have dropped circle CI test. But I have added bash script to setup `vault` environment easily.

I have noticed `vault` seems pause when dealing with `expired` logins during integration test, not sure if you have faced the same issue during your test.

wgx731 committed:
refactor[core,manage,auth]: setup

1. setup style check
2. setup `startVaultTask`
3. update tests

wgx731 committed:
fix[auth,core,manage] build.sbt update

only start one vault server

wgx731 committed:
test[core] update test based on AppRole

wgx731 committed:
style[core,manage,auth] use ⇒

wgx731 committed:
chore[core,manage,auth] rm stopped container

wgx731 committed:
chore[core,manage,auth] update libs

wgx731 committed:
style[core,manage,auth] reformat test

wgx731 committed:
chore[core,manage,auth] update circle settings

wgx731 committed:
chore[core,manage,auth] update circle system

wgx731 committed:
Revert

"chore[core,manage,auth] update circle settings"

This reverts commit dd286e4fe2fc238ac94ee33a411d554a85986834.

wgx731 committed:
test[manage]: fix inspect policy

NOTE: return from vault of policy is not valid Json

wgx731 committed:
test[manage]: fix mount list error

wgx731 committed:
chore[auth]: remove Token part

wgx731 committed:
doc[core,manage,auth]: update README

wgx731 committed:
chore[core,manage,auth]: remove circle CI

wgx731 committed:
feat[core,manage,auth]: update version

janstenpickle commented:
@wgx731 Thanks for doing this, I had just started converting to using App Roles and upgrading vault etc, so you've saved me loads of time! The house keeping and adding scalastyle is really good, so thanks for that as well.

I'll try and get this reviewed ASAP so I can do a release and unblock everyone. I'll have a look into a new CI system to test this so I can verify future pull requests.

janstenpickle commented on auth/src/it/scala/janstenpickle/vault/auth/UserPassIT.scala:
```diff
@@ -30,37 +30,37 @@ class UserPassIT extends VaultSpec with ScalaCheck {
     authAdmin.disable(client).attemptRun(_.getMessage()) must beOk
 
-  def authPass = test((username, password, client, ttl) =>
+  def authPass = test((username, password, client, ttl) ⇒
```
I don't really like the use of `⇒` over `=>`. I realise this comes down to a matter of preference, but a style change like this should really be discussed before including in a pull request with feature code :)

wgx731 replied:
done

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -13,8 +12,10 @@ import uscala.result.Result.{Fail, Ok}
+  )
 
 
   import cats.Applicative
```
Oops, I managed to commit some play code for doing something else here, I'll remove it.

wgx731 replied:
I will leave this part as it is so you can decide what to change.

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
 }
 
-object option {
+object CatsOption {
```
The reason for calling the object above here `CatsConversion` was for converting from cats types to uscala, here the code is just providing syntax for `Option`, so can just be called `Option` rather than `CatsOption`.

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
-object asyncresult {
-  implicit class FutureToAsyncResult[T](future: Future[T])(implicit ec: ExecutionContext) {
-    def toAsyncResult: AsyncResult[String, T] = AsyncResult(future.map(Result.ok))
+object CatsAsyncResult {
```
As above, please rename this to `AsyncResult`.

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
 }
 
-object vaultconfig {
+object CatsVaultConfig {
```
As above, please rename this to `VaultConfig`

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
 
-object json {
-  import catsconversion._
+object CatsJson {
```
As above, please rename this to `Json`

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
-object response {
-  import catsconversion._
-  import json._
+object CatsResponse {
```
As above, please rename to `Response`

wgx731 replied:
@janstenpickle, the reason I have add the `Cats` prefix is because compile error like:
```
[error] /home/eric/Work/try-out/scala/scala-vault/core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:3: imported `Response' is permanently hidden by definition of object Response in package syntax
[error] import com.ning.http.client.Response 
```
Please let me know if there is better ways to deal with the error above.

And I think it's more about style of naming, we can always change scalastyle config to allow lowercase class name. 

Thanks

janstenpickle commented on core/src/main/scala/janstenpickle/scala/syntax/syntax.scala:
```diff
@@ -24,95 +25,122 @@ object catsconversion {
 }
 
-object request {
+object CatsRequest {
```
As above, please rename to `Request`

wgx731 committed:
doc[core,manage,auth]: add testing command

wgx731 committed:
style[core,manage,auth]: change to `=>`

wgx731 committed:
feat[manage]: add back list mount model

wgx731 commented:
@janstenpickle, thanks for the quick feedback. Please refer to my comments on the reviews. Besides that, as I am still learning `scala`, `cats` and `circe`, would like to have your input on the following areas:
* [HCursor => ACursor](https://github.com/wgx731/scala-vault/blob/dev/core/src/main/scala/janstenpickle/scala/syntax/syntax.scala#L95): not sure using `_.downArray` is a good choice here or not.
* [Mount optional config object](https://github.com/wgx731/scala-vault/blob/dev/manage/src/main/scala/janstenpickle/vault/manage/manage.scala#L101): as [vault doc](https://www.vaultproject.io/api/system/mounts.html#parameters) says,  the object now has 3 possible values and the config object can be optional as well I think. What's the best way to deal with this?
* [Policy inspect](https://github.com/wgx731/scala-vault/blob/dev/manage/src/main/scala/janstenpickle/vault/manage/manage.scala#L72): from [vault doc](https://www.vaultproject.io/api/system/policy.html#sample-response-1), the response from `vault` is not valid json, therefore I have changed to return `String` for now, but we may need to think a better way to handle this, or is it a bug from `vault`? 😃 
* [`Token` in `auth` module](https://github.com/wgx731/scala-vault/commit/2b9eb87ebc6ad38ac505cbb41f9305d74472acb9), as I am not really familiar with `Token` class and due to changes from `vault` all test seem failing, therefore I have removed it for now, please add back when you have time.

By the way, please help me to check if the [test setup](https://github.com/wgx731/scala-vault/tree/dev#testing) is ok, as I only tested on my machine. Please note that, if you are on ubuntu and use `aufs` storage driver for docker, you may sometimes facing [this issue](https://github.com/hashicorp/docker-vault/issues/24), just restart docker and wait for some time before return test again then it should be ok.

By the way, do you think having a `dev` branch is a good idea? 

Thank you very much.

janstenpickle commented:
@wgx731 thanks again, this is really good stuff. I still haven't had time to check out your branch, test and build it, however in regard to your points

* [HCursor => ACursor](https://github.com/wgx731/scala-vault/blob/dev/core/src/main/scala/janstenpickle/scala/syntax/syntax.scala#L95) - I'm not sure either, I'd need to look more deeply into circe to see what it's doing, I suppose the important question is does the change break the tests?
* [Mount optional config object](https://github.com/wgx731/scala-vault/blob/dev/manage/src/main/scala/janstenpickle/vault/manage/manage.scala#L101): The best way to deal with this would be to add a new optional field to the `MountConfig` case class to represent the new config setting
*  [Policy inspect](https://github.com/wgx731/scala-vault/blob/dev/manage/src/main/scala/janstenpickle/vault/manage/manage.scala#L72): There was code to cope with non-json rules, unless this has stopped working I suggest [leaving what was there before](https://github.com/janstenpickle/scala-vault/blob/master/manage/src/main/scala/janstenpickle/vault/manage/manage.scala#L65). I do admit, it's very annoying that they aren't valid JSON
* [`Token` in `auth` module](https://github.com/wgx731/scala-vault/commit/2b9eb87ebc6ad38ac505cbb41f9305d74472acb9): It should be fine to remove this for now

With regards to a dev branch, I agree it's probably a good idea, after this has been merged I'll probably turn on git-flow.

With regards to the naming of the syntax classes, probably best append them with `Syntax`, which will avoid clashes with other imports.

I'll try and look at this properly tomorrow, if you have any other questions or need me to be clearer just let me know.

wgx731 committed:
refactor[core,manage,auth]: change `Cats` prefix

wgx731 commented:
@janstenpickle, changed the class prefix to `Syntax`, test should be all passing. Please let me know if I can help with any other parts, looking forward to a newer version release.

Thank you very much.

Commit:
Add travis and some renaming

add travis config for CI with docker
change naming for syntax classes
update mount config to include new setting

janstenpickle commented:
@wgx731 merged, great job 😄. Once the build is done on master I'll cut a release.
